pkg/vsphere/simulator/property_collector.go

```diff
@@ -40,6 +41,37 @@ func NewPropertyCollector(ref types.ManagedObjectReference) object.Reference {
 var errMissingField = errors.New("missing field")
 var errEmptyField = errors.New("empty field")
 
+func getObject(ref types.ManagedObjectReference) (reflect.Value, bool) {
+	obj := Map.Get(ref)
+	if obj == nil {
+		return reflect.Value{}, false
+	}
+
+	rval := reflect.ValueOf(obj).Elem()
+	rtype := rval.Type()
+
+	// PropertyCollector is for Managed Object types only (package mo).
+	// If the registry object is not in the mo package, assume it is a wrapper
+	// type where the first field is an embedded mo type.
+	// We need to dig out the mo type for PropSet.All to work properly and
+	// for the case where the type has a field of the same name, for example:
+	// mo.ResourcePool.ResourcePool
+	for {
+		if path.Base(rtype.PkgPath()) != "mo" {
+			if rtype.Kind() != reflect.Struct || rtype.NumField() == 0 {
+				log.Printf("%#v does not have an embedded mo type", ref)
+				return reflect.Value{}, false
+			}
+			rval = rval.Field(0)
```

Comment:
Yes, it already works in similar cases such as with `ClusterComputeResource` which extends `ComputeResource`.